Neos.TimeableNodeVisibility/Classes/Domain/HandlingResultSet.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Neos\TimeableNodeVisibility\Domain;
+
+class HandlingResultSet
+{
+    /**
+     * @var HandlingResult[]
+     */
+    private array $handlingResults = [];
+
+    public function __construct()
+    {
+    }
+
+    public function add(HandlingResult $handlingResult)
+    {
+        $this->handlingResults[] = $handlingResult;
+    }
```

Comment:
I would suggest to make this collection immutable, as we are more used to this code style with neos 9 and never code. It seems the collection is only build in one place at the `TimeableNodeVisibilityService` so we could just instead use a temporary array and then spread the results in the constructor:

(also i think our convention - at least by looking at existing code - is to identify collections either by plural or `Collection` suffix.

```suggestion
final readonly class HandlingResultCollection # or HandlingResults
{
    /**
     * @var HandlingResult[]
     */
    private array $handlingResults = [];

    public function __construct(HandlingResult ...$handlingResults)
    {
        $this->handlingResults = $handlingResults;
    }
```